src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        try {
+            LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+            LocalDate dateFinish = LocalDate.parse(dateTo, formatter);
+            int salary = 0;
+            builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+            for (String name : names) {
+                for (String str : data) {
+                    String[] values = str.split(" ");
+                    LocalDate dateWork = LocalDate.parse(values[0], formatter);
+                    if (dateWork.isAfter(dateStart.minusDays(1))
+                            && dateWork.isBefore(dateFinish.plusDays(1))
+                            && name.equals(values[1])) {
```

Comment:
1) Pay attention to the order of expressions in this condition. If the current "name" is John for example, then first of all you should check if the current "values" refer to John. You should first check the name and only then create dateWork and validate the date.
2) It's hard to catch up with what happens "name.equals(values[1])" and what is stored in "values[1]" so it's better to extract this expression to a separate variable with a proper name so it gives more context